### ISSUE TITLE
`install({ toFake: ["setImmediate"] })` fails when it's not available in context

### ISSUE THREAD

**Issue by hi-ogawa** [sinonjs/fake-timers#490]
-   FakeTimers version : v11.2.2
-   Environment : reproduction is in stackblitz nodejs, but it's likely in any environment since reproduction uses `withGlobal` to pass context manually
-   Example URL : https://stackblitz.com/edit/vitest-dev-vitest-yzjgau?file=repro-setImmediate.mjs
-   Other libraries you are using: FakeTimers is only dependency in reproduction

**What did you expect to happen?**

No error. Preferably allow faking `setImmediate` even when it's not available in the context, which would align with current behavior with `requestIdleCallback` (see `repro-requestIdleCallback.mjs` in reproduction).

**What actually happens**

```
Error: Cannot set properties of undefined (setting 'hadOwnProperty')
```

**How to reproduce**

```ts
import { withGlobal } from '@sinonjs/fake-timers';

function main() {
  const global = { Date, setTimeout, clearTimeout };
  const timers = withGlobal(global);

  // error when install
  // > Error: Cannot set properties of undefined (setting 'hadOwnProperty')
  const clock = timers.install({ toFake: ['setImmediate'] });
}

main();
```

### extra context of the issue

First of all, thanks for making this portable library for timer mocking!
I found some inconsistency of `@sinonjs/fake-timers` while investigating an issue on Vitest https://github.com/vitest-dev/vitest/issues/5027#issuecomment-1903570906. There is a workaround on Vitest side so it's not critical, but I thought it's worth reporting an issue.

This is likely a same issue as https://github.com/sinonjs/fake-timers/issues/277 but I think the issue still persists since the verification https://github.com/sinonjs/fake-timers/issues/277#issuecomment-851504314 seems to be done on NodeJS where `setImmediate` is globally available but this is not the case on browsers.

I'm not sure what's desired behavior, but there seems slight inconsistency whether `@sinonjs/fake-timers` fakes what's not globally available. From what I tried so far, `{ toFake: ["requestIdleCallback"] }` works even when it's not globally available (for example, this is browser-only and not available on NodeJS). On the other hand, `{ toFake: ["setImmediate"] }` throws an error on browser.

**Comment by SimenB** [comment]
Failing with an explicit error in either case seems reasonable, but that might be considered a breaking change?

**Comment by fatso83** [comment]
> Failing with an explicit error in either case seems reasonable, but that might be considered a breaking change?

Absolutely, but I would also add a flag to silence those errors to have an easy fallback to the previous behaviour. That way Jest could consume `fake-timers@12` by passing `ignoreMissingTimers: true` (just a suggestion) by default.

**Comment by fatso83** [comment]
Added the breaking change in #491 with a flag to ignore missing. That should make the behaviour consistent going forwards and maybe make introducing #323 simpler. Took a lot more effort to get the testcases working than expected, but at least it is now working.

Have a look, otherwise I'll just merge in a few days.
